internal/document.go

```diff
@@ -54,3 +55,189 @@ func (doc Document) Debug() (bool, error) {
 	}
 	return true, nil
 }
+
+func (doc Document) Verify(
```

Comment:
I pulled all of the document related "verify" steps from `internal/nitrite.Verify()`. Furthermore, I divided these steps into three chunks (checking mandatory fields, optional fields, and certificates) for the sake of readability and testing.